### PR TITLE
XAeroNet

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Description

PR for adding the XAeroNet model for scalable external vehicle aerodynamics

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Review by Alexey-Kamenev — COMMENTED** [review]
Some minor comments, still reviewing.

File: examples/cfd/xaeronet/README.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# XAeroNet: Scalable Foundational Neural Models for External Aerodynamics
+
+XAeroNet is a collection of scalable foundational models for large-scale external
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/cfd/xaeronet/README.md:3]: Thanks. I removed this term as the model is not a foundation model (yet).

File: examples/cfd/xaeronet/README.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+aerodynamic evaluatons. It consists of two models, XAeroNet-S and XAeroNet-V for
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/cfd/xaeronet/README.md:4]: Fixed it

File: examples/cfd/xaeronet/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+sruface and volume predictions, respectively.
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/cfd/xaeronet/README.md:5]: Fixed it

File: examples/cfd/xaeronet/volume/train.py [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+            project="aws_drivaer",
+            entity="Modulus",
+            name="aws_drivaer",
+            mode="disabled",
+            group="group",
+            save_code=True,
+        )
+
+    # AMP Configs
+    amp_dtype = torch.bfloat16  # UNet does not work with bfloat16
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/cfd/xaeronet/volume/train.py:93]: Oh Thanks for noticing this. The comment is correct but the code is not. I changed it to float16.

File: examples/cfd/xaeronet/surface/train.py [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+    # Instantiate the distributed manager
+    DistributedManager.initialize()
+    dist = DistributedManager()
+    device = dist.device
+    print(f"Rank {dist.rank} of {dist.world_size}")
+
+    # Instantiate the writers
+    if dist.rank == 0:
+        writer = SummaryWriter(log_dir="tensorboard")
+        initialize_wandb(
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/cfd/xaeronet/surface/train.py:77]: This is a placeholder for now and I'll add wb loggings in future PRs. The volume model already has wb loggings.

File: examples/cfd/xaeronet/surface/train.py [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+                    target = torch.cat(
+                        (part.ndata["pressure"], part.ndata["shear_stress"]), dim=1
+                    )
+                    target_filtered = target[part.ndata["inner_node"].bool()]
+                    loss = (
+                        torch.mean((pred_filtered - target_filtered) ** 2)
+                        / cfg.num_partitions
+                    )
+                    total_loss += loss.item()
+                scaler.scale(loss).backward()
```
Comment (Alexey-Kamenev) [inline examples/cfd/xaeronet/surface/train.py:200]: I was wondering since this is similar to gradient accumulation, would it make sense to use DDP `no_sync` here to improve the perf?

File: examples/cfd/xaeronet/surface/train.py [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+        input_dim_nodes=6,
+        input_dim_edges=4,
+        output_dim=4,
+        processor_size=cfg.num_message_passing_layers,
+        aggregation="sum",
+        hidden_dim_node_encoder=256,
+        hidden_dim_edge_encoder=256,
+        hidden_dim_node_decoder=256,
+        mlp_activation_fn="relu",
+        do_concat_trick=True,
```
Comment (Alexey-Kamenev) [inline examples/cfd/xaeronet/surface/train.py:145]: Use value from config? Same for `num_processor_checkpoint_segments` - there is already a setting in `config.yaml`.

**Review by Alexey-Kamenev — APPROVED** [review]

File: examples/cfd/xaeronet/surface/train.py (hunk 7 above)
Comment (mnabian) [reply examples/cfd/xaeronet/surface/train.py]: Yes I fixed this but forgot to push. Please see the last commit.

File: examples/cfd/xaeronet/surface/train.py (hunk 6 above)
Comment (mnabian) [reply examples/cfd/xaeronet/surface/train.py:211]: I tried this but unfortunately I can't get it to work properly with AMP. I'll explore this further and open another PR once I fix the issue with AMP.
